app/models/docker_builder_service.rb

```diff
@@ -60,6 +61,7 @@ def push_image(tag)
         build.docker_repo_digest = "#{project.docker_repo}@#{matches[1]}"
       end
     end
+    build.docker_repo_digest ||= result.info['RepoTags'].first
```

Comment:
It looks like Docker Hub is not sending back a digest when pushing an image. Unlike our private registry which does.
So I fall back to a tagged version in case it didn't.